core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexUpgradeService.java

```diff
@@ -77,7 +77,12 @@ public IndexMetaData upgradeIndexMetaData(IndexMetaData indexMetaData) {
         IndexMetaData newMetaData = indexMetaData;
         // we have to run this first otherwise in we try to create IndexSettings
         // with broken settings and fail in checkMappingsCompatibility
-        newMetaData = archiveBrokenIndexSettings(newMetaData);
+        try {
+            newMetaData = archiveBrokenIndexSettings(newMetaData);
+        } catch (Exception ex) {
+            logger.error("{} failed to process index settings: {}", newMetaData.getIndex(), ex.getMessage());
```

Comment:
this would be quite a bit of refactoring and the settings validation is not index specific. I am very close to just close this issue and leave it as it is. It's such a corner case and all I tried to prevent is failing late. Even further the bloody index isn't even important you gotta set the node limit anyway...